flink-connectors/flink-connector-kinesis/src/main/java/org/apache/flink/streaming/connectors/kinesis/internals/publisher/RecordPublisherFactory.java

```diff
@@ -37,6 +46,41 @@
 	 * @param streamShardHandle the stream shard in which to consume from
 	 * @return the constructed {@link RecordPublisher}
 	 */
-	RecordPublisher create(Properties consumerConfig, MetricGroup metricGroup, StreamShardHandle streamShardHandle);
+	RecordPublisher create(
+			SequenceNumber sequenceNumber,
+			Properties consumerConfig,
+			MetricGroup metricGroup,
+			StreamShardHandle streamShardHandle) throws InterruptedException;
+
+	/**
+	 * Determines the starting position in which the {@link RecordPublisher} should start consuming from.
+	 *
+	 * @param sequenceNumber the sequence number to start from
+	 * @param consumerConfig the consumer properties
+	 * @return the {@link StartingPosition} in which to start consuming from
+	 */
+	default StartingPosition getStartingPosition(
```

Comment:
Was there any particular reason to put this method in the `RecordPublisherFactory`, instead of letting the factory get a `StartingPosition` (and not a `SequenceNumber`) on `create`?
The conversion from sequence number to `StartingPosition` logic could probably be a separate static utility method somewhere that is unrelated to the actual implementation of the `RecordPublisherFactory`.

Reply:
No reason in particular, I am happy to move it. I am thinking a good place could be in `StartingPosition` itself, as we already have `StartingPosition.restartFromSequenceNumber(SequenceNumber)`. I could add  `StartingPosition.restartFromSequenceNumber(SequenceNumber, Properties)`, although it does feel like that class should not know how to parse properties, it might be better than creating a new class for that single method. 

I will go with this and can change it if you disagree.

Reply:
That sounds good to me, thanks.
If we want to be extra explicit, perhaps only the required properties are parsed and then passed in to `StartingPosition.restartFromSequenceNumber`

Reply:
👍

Reply:
It didn't feel right to pass the `@Nullable` fields to `StartingPosition` so in the end:
- `Date` is parsed from properties in `KinesisConfigUtil`
- Logic to determine to restart from timestamp or sequence is in `AWSUtil`